### DISCUSSION TITLE
Unable to Open Jupyter Notebooks

### DISCUSSION THREAD

**Discussion by gogasca** [eclipse-theia/theia#9111] [Q&A] [answered]
### Bug Description:

Unable to display Jupyter Notebooks (.ipynb) files using "non localhost" in the Web browser.
I get this error:

```
The webpage at http://30d3ad54-bb26-40b1-801d-76a8015cf154.webview.127.0.0.1/webview/index.html?id=30d3ad54-bb26-40b1-801d-76a8015cf154 might be temporarily down or it may have moved permanently to a new web address.
```
Works: Use **localhost** in browser
Does not work: Using **127.0.0.1** or my **Proxy** address. I always get the same error. I can access other files and Theia UI, only notebooks fail to display. 

### Steps to Reproduce:

0. Run Thea as container
```
docker run -it -p 127.0.0.1:80:8080 -v "$PWD:/tmp"  gcr.io/<my_project>/<container_name>:latest
```
1. Access browser using 127.0.0.1 in the address
2. Open a Jupyter Notebook file (.ipynb)
3. You will get an error:

```
The webpage at http://30d3ad54-bb26-40b1-801d-76a8015cf154.webview.127.0.0.1/webview/index.html?id=30d3ad54-bb26-40b1-801d-76a8015cf154 might be temporarily down or it may have moved permanently to a new web address.
```

![ipynb_error](https://user-images.githubusercontent.com/30065079/108810280-a83f5d80-755f-11eb-9bdd-6415f47ed5cb.png)

### Additional Information

- Operating System: MacOS
- Theia Version:

```
@theia/callhierarchy 1.10.0
@theia/console 1.10.0
@theia/core 1.10.0
@theia/cpp-debug 1.0.0
@theia/debug 1.10.0
@theia/editor 1.10.0
@theia/editor-preview 1.10.0
@theia/file-search 1.10.0
@theia/filesystem 1.10.0
@theia/getting-started 1.10.0
@theia/git 1.10.0
@theia/keymaps 1.10.0
@theia/markers 1.10.0
@theia/messages 1.10.0
@theia/metrics 1.10.0
@theia/mini-browser 1.10.0
@theia/monaco 1.10.0
@theia/navigator 1.10.0
@theia/outline-view 1.10.0
@theia/output 1.10.0
@theia/plugin-ext 1.10.0
@theia/plugin-ext-vscode 1.10.0
@theia/preferences 1.10.0
@theia/preview 1.10.0
@theia/process 1.10.0
@theia/scm 1.10.0
@theia/scm-extra 1.10.0
@theia/search-in-workspace 1.10.0
@theia/task 1.10.0
@theia/terminal 1.10.0
@theia/timeline 1.10.0
@theia/typehierarchy 1.10.0
@theia/userstorage 1.10.0
@theia/variable-resolver 1.10.0
@theia/vsx-registry 1.10.0
@theia/workspace 1.10.0
```

This is what I see in my Browser (Chrome Version 88.0.4324.150)

Code in browser when it fails:

```
<iframe class="webview" sandbox="allow-scripts allow-forms allow-same-origin" src="http://fecae9aa-9e56-4675-adff-5984091b3302.webview.127.0.0.1/webview/index.html?id=fecae9aa-9e56-4675-adff-5984091b3302" style="border: none; width: 100%; height: 100%;"></iframe>
```

Is sandbox configuration "allow-same-origin" the problem?:
```
https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/main/browser/webview/webview.ts#L250
```

I tried setting:

```
ENV THEIA_WEBVIEW_EXTERNAL_ENDPOINT={{uuid}}.webview.{{localhost}}
```

but same issue. Any pointers will be appreciated.?

**Comment by vince-fugnitto** [comment, marked as answer]
@gogasca thank you for the discussion, I migrated the issue to a discussion since it is more of a question.
I believe you may want to try to setup your environment correctly, there are a couple of issues that describe how to do so:
- https://github.com/eclipse-theia/theia/pull/6465#issuecomment-557494879
- https://community.theia-ide.org/t/images-either-open-as-text-or-have-incorrect-url/1513/2
- https://community.theia-ide.org/t/theia-webviews-and-virtual-hosting/1066
